src/app/components/normal-component/count-down/count-down.component.ts

```diff
@@ -45,9 +47,20 @@ export class CountDownComponent implements OnInit, OnDestroy {
         this.changeDuration(duration);
     }
     public changeDuration(ms: number): void {
+        let mustResume: boolean = false;
+        if (this.isPaused === false) {
+            this.pause();
+            mustResume = true;
+        }
```

Comment:
hm pourquoi le mettre en pause ?
Simplement faire `remainingMsToDisplay = this.remainingMS`, puis calculer `this.displaySec` et `this.displayMinute` ne suffit pas ?

Reply:
Euh, la pause puis resume permet de pauser le timeout et puis de le reprendre (la pause calculant elle même à quel ms précis le chrono était, vu qu'il met pas à jours les ms chaque ms !) ça répond à la question ?

Reply:
ook ça fait plus ou moins sens :D Tu peux résoudre !